src/fsharp/FSharp.Core/array.fsi

```diff
@@ -459,6 +459,26 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("ForAll2")>]
         val forall2: predicate:('T1 -> 'T2 -> bool) -> array1:'T1[] -> array2:'T2[] -> bool
 
+        /// <summary>Tests if all corresponding elements of three arrays satisfy the given predicate pairwise.</summary>
+        ///
+        /// <remarks>The predicate is applied to matching elements in the three collections up to the lesser of the 
+        /// three lengths of the collections. If any application returns false then the overall result is 
+        /// false and no further elements are tested. Otherwise, if one collection is longer 
+        /// than the others then the <c>ArgumentException</c> exception is raised. 
+        /// Otherwise, true is returned.</remarks>
+        ///
+        /// <param name="predicate">The function to test the input elements.</param>
+        /// <param name="array1">The first input array.</param>
+        /// <param name="array2">The second input array.</param>
+        /// <param name="array3">The third input array.</param>
+        ///
+        /// <exception cref="T:System.ArgumentNullException">Thrown when any of the input arrays is null.</exception>
+        /// <exception cref="T:System.ArgumentException">Thrown when the input arrays differ in length.</exception>
+        ///
+        /// <returns>True if all of the array elements satisfy the predicate.</returns>
+        [<CompiledName("ForAll3")>]
+        val internal forall3: predicate:('T1 -> 'T2 -> 'T3 -> bool) -> array1:'T1[] -> array2:'T2[] -> array3:'T3[] -> bool
```

Comment:
To be honest I'd rather you didn't add an internal function to this API.  Just put this in Linq.fs - probably best as a direct loop over the array index with a mutable